### PR TITLE
feat: add German language support

### PR DESCRIPTION
**Pull request by chrisschuller**

### CODE REVIEW

**Comment by chrisschuller** [comment]
This is more professional sounding German language support than the pull request that was created 9h earlier and that I was not aware of at the time. 😉

File: messages/de-DE.json [hunk 1]
```diff
@@ -0,0 +1,384 @@
...
+    "title": "Ausgaben",
+    "description": "Hier sind die Ausgaben, die du für deine Gruppe erstellt hast.",
+    "create": "Ausgabe hinzufügen",
+    "createFirst": "Erstelle die Erste",
+    "noExpenses": "Deine Gruppe hat noch keine Ausgaben.",
+    "exportJson": "Als JSON exportieren",
+    "searchPlaceholder": "Suche nach einer Ausgabe…",
+    "ActiveUserModal": {
+      "title": "Wer bist du?",
+      "description": "Sag uns, welcher Teilnehmer du bist, damit wir die Informationen, die wir dir zeigen, auf dich zuschneiden können.",
```
Comment (tedzards509) [inline messages/de-DE.json:19]: Weirdly convoluted in my opinion. I would rephrase to "Sag uns, welcher Teilnehmer du bist, um die angezeigten Informationen auf dich anzupassen." or similar.

File: messages/de-DE.json [hunk 2]
```diff
@@ -0,0 +1,384 @@
...
+      "placeholder": "Sommerurlaub",
+      "description": "Gib deiner Gruppe einen Namen."
+    },
+    "InformationField": {
+      "label": "Gruppeninformationen",
+      "placeholder": "Welche Informationen sind relevant für Gruppenmitglieder?"
+    },
+    "CurrencyField": {
+      "label": "Währungssymbol",
+      "placeholder": "$, €, £…",
```
Comment (tedzards509) [inline messages/de-DE.json:84]: I think in German localization, € coming before $ would make sense.

File: messages/de-DE.json [hunk 3]
```diff
@@ -0,0 +1,384 @@
...
+      "add": "Mitglied hinzufügen",
+      "John": "Johannes",
+      "Jane": "Janina",
+      "Jack": "Jakob"
+    },
+    "Settings": {
+      "title": "Lokale Einstellungen",
+      "description": "Dies sind Einstellungen pro Gerät, die verwendet werden, um deine Nutzungsexperience zu verbessern.",
+      "ActiveUserField": {
+        "label": "Aktiver nutzer",
```
Comment (tedzards509) [inline messages/de-DE.json:101]: Nutzer should be capital.

File: messages/de-DE.json [hunk 4]
```diff
@@ -0,0 +1,384 @@
...
+      },
+      "categoryFieldDescription": "Wähle die Kategorie der Einnahme.",
+      "paidByField": {
+        "label": "Empfangen von",
+        "description": "Wähle das Mitglied, das die Einnahme erhalten hat."
+      },
+      "paidFor": {
+        "title": "Empfangen für",
+        "description": "Wähle für wen die Einnahme empfangen wurde."
+      },
```
Comment (tedzards509) [inline messages/de-DE.json:134]: This doesn't really make sense to me. I think it should be rephrased to make it clearer who received money and who should get a split. On the other hand, I don't even know where in the app these strings appear.

File: messages/de-DE.json [hunk 5]
```diff
@@ -0,0 +1,384 @@
...
+      "splitModeDescription": "Wähle, wie die Einnahme aufgeteilt werden soll.",
+      "attachDescription": "Sehe und füge einen Beleg für die Einnahme hinzu."
```
Comment (tedzards509) [inline messages/de-DE.json:136]: "Sehe und füge hinzu"? I don't know what that is supposed to mean.

Also "füge der Einnahme einen Beleg hinzu" sounds more elegant to me.

File: messages/de-DE.json [hunk 6]
```diff
@@ -0,0 +1,384 @@
...
+    "create": "Erstellen",
+    "creating": "Erstellt…",
+    "save": "Speichern",
+    "saving": "Speichert…",
+    "cancel": "Abbrechen"
+  },
+  "ExpenseDocumentsInput": {
+    "TooBigToast": {
+      "title": "Das File ist zu groß",
+      "description": "Die maximale File-Größe ist {maxSize}. Deine ist ${size}."
```
Comment (tedzards509) [inline messages/de-DE.json:203]: "Datei" and "maximale Dateigröße"

File: messages/de-DE.json [hunk 7]
```diff
@@ -0,0 +1,384 @@
...
+    },
+    "ErrorToast": {
+      "title": "Fehler beim Hochladen des Dokuments",
+      "description": "Beim Hochladen des Dokuments ist etwas schiefgelaufen. Versuche es später nochmal oder wähle ein anderes File.",
```
Comment (tedzards509) [inline messages/de-DE.json:207]: "... oder wähle eine andere Datei"

File: messages/de-DE.json [hunk 8]
```diff
@@ -0,0 +1,384 @@
...
+      "editNext": "Als nächstes kannst du die Informationen zur Ausgabe editieren.",
+      "continue": "Weiter"
+    },
+    "unknown": "Unbekannt",
+    "TooBigToast": {
+      "title": "Das File ist zu groß",
+      "description": "Die maximale File-Größe ist {maxSize}. Deine ist ${size}."
+    },
+    "ErrorToast": {
+      "title": "Fehler beim Hochladen des Dokuments",
```
Comment (tedzards509) [inline messages/de-DE.json:231]: Here I would also use "Datei" instead of document to be more uniform.

File: messages/de-DE.json [hunk 9]
```diff
@@ -0,0 +1,384 @@
...
+      "retry": "Wiederholen"
+    }
+  },
+  "Balances": {
+    "title": "Bilanz",
+    "description": "Das sind die Beträge, die jedes Mitglied bezahlt oder empfangen hat.",
+    "Reimbursements": {
+      "title": "Vorgeschlagene Rückzahlungen",
+      "description": "Hier sind Vorschläge für optimierte Rückzahlungen zwischen Mitgliedern.",
+      "noImbursements": "Es sieht aus, als bräuchte deine Gruppe keine Rückzahlungen 😁",
```
Comment (tedzards509) [inline messages/de-DE.json:242]: The group doesnt need the reimbursements at all. So "als seien in der Gruppe keine Rückzahlungen nötig" is a better way to phrase it in my view as it looks at the group members, not the group itself.

**Review by tedzards509 — CHANGES_REQUESTED** [review]
I agree that this translation is better that the older one. I have some issues woth this though.

File: messages/de-DE.json (hunk 9 above)
Comment (chrisschuller) [reply messages/de-DE.json:242]: I wanted to stay closer to the English translation but your version sounds better :)

File: messages/de-DE.json (hunk 4 above)
Comment (chrisschuller) [reply messages/de-DE.json:134]: Yeah I also have not found these strings in the app. Therefore I would like to leave the translation as-is, until I can verify that it doesn't match the semantic context.

File: messages/de-DE.json (hunk 9 above)
Comment (Sunsurfer89) [reply messages/de-DE.json:242]: If German, we should do it properly. "Denglish" might not look professional. I fully agree with the comments and proposals of @tedzards509

**Comment by chrisschuller** [comment]
@tedzards509 I have added a new commit in which I integrated almost all your recommendations. Thank you! :)

**Review by Sunsurfer89 — COMMENTED** [review]
Just adding the cherry on the cake, rest is fine so far. 👍

**Comment by chrisschuller** [comment]
@Sunsurfer89 Thank you for your recommendations. I added them to the PR. I would hope this PR gets merged soon. I like the project very much, but the time for merging PRs seems to be a bit slow, sadly. :)

**Comment by chrisschuller** [comment]
@scastiel This PR seems to be complete. I think the German user base would appreciate the added language support. Would you be so kind as to merge it soon? :)

**Comment by chrisschuller** [comment]
@scastiel I think it could also make sense to add translations for the home page and the page's meta data, like the title for example. Should we, in a follow-up PR, add translation keys for this as well?

**Comment by Sunsurfer89** [comment]
> @Sunsurfer89 Thank you for your recommendations. I added them to the PR. I would hope this PR gets merged soon. I like the project very much, but the time for merging PRs seems to be a bit slow, sadly. :)

I hope so, too. I have a use case in September and the other contributors to my expense group do not have the best English language skills... so any German translation would be great. And I really love this project, too. So, I hope for a quick release!

**Comment by max298** [comment]
Thanks for this PR! As a native German I really appreciate your efforts. I can also confirm that these translations are really good 👍 

Not trying to add additional delay, but shouldn't the files be called `ger-GER` (as in **ger**man) instead of `de-DE` like **de**utsch in order to follow the existing schema of the other files (like `fi.json` and also the other open PRs with `es.json` etc.)?

**Comment by chrisschuller** [comment]
I would think that the current naming follows the ISO standard for German language in Germany. This is important as there might be different translations for German speakers in Austria or Switzerland :)

**Comment by Sunsurfer89** [comment]
@max298 nope: DE-de is correct! Check out: [https://en.m.wikipedia.org/wiki/ISO_3166-1_alpha-2](https://en.m.wikipedia.org/wiki/ISO_3166-1_alpha-2)

We have this discussion also far too often at work. ISO is the answer 😉

**Comment by scastiel** [comment]
Amazing, thanks @chrisschuller for the translation, and thanks all for the reviews, you rock ❤️
